pennylane/beta/qnode.py

```diff
@@ -0,0 +1,540 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the QNode class and qnode decorator.
+"""
+# pylint: disable=too-many-instance-attributes,too-many-arguments,protected-access
+from collections.abc import Sequence
+import functools
+import inspect
+import warnings
+
+import pennylane as qml
+from pennylane import Device
+from pennylane.interfaces.batch import set_shots, SUPPORTED_INTERFACES
+
+
+class QNode:
+    """Represents a quantum node in the hybrid computational graph.
+

```

Comment:
Good call :) I'll add a `.. warning::` block.

```suggestion

    .. warning::
    
        This QNode is a beta feature. It differs from the standard :class:`~.pennylane.QNode`
        in several ways:
        
        - Custom gradient transforms can be specified as the differentiation method.

        - Arbitrary :math:`n`-th order derivatives are supported on hardware using
          gradient transforms such as the parameter-shift rule.

        - Internally, if multiple circuits are generated for execution simultaneasouly, they
          will be packaged into a single job for execution on the device. This can lead to
          significant performance improvement when executing the QNode on remote
          quantum hardware.
          
        In an upcoming release, this QNode will replace the existing one. If you come across any
        bugs while using this QNode, please let us know via a `bug report
        <https://github.com/PennyLaneAI/pennylane/issues/new?assignees=&labels=bug+%3Abug%3A&template=bug_report.yml&title=%5BBUG%5D`__ on our GitHub bug tracker.
        
        Currently, this beta QNode does not support the following features:
        
        - Circuit decompositions
        - Non-mutabability via the `mutable` keyword argument
        - Viewing specifications with `qml.specs`
        
        It is also not tested with the :mod:`~.qnn` module.
```

Reply:
I think you accidentally resolved the conversation before committing the suggestion here :sweat_smile:

Reply:
Wait, I can see this new snippet in the files changed tab?

Reply:
https://pennylane--1642.org.readthedocs.build/en/1642/code/api/pennylane.beta.qnode.html

Reply:
Yeah it was super weird, because the section was even rendering in the docs, it just wasn't showing up in the "files changed" for me :confused: